### PR TITLE
Cloud provider snapshot restore job

### PR DESCRIPTION
**Pull request by coderGo93**
Added a resource for cloud provider snapshot restore jobs, now it can create/delete(cancel) a cloud provider snapshot restore job

### CODE REVIEW

File: cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+		}
+		if targetProjectId == nil || *targetProjectId == "" {
+			return handler.ProgressEvent{
+				OperationStatus: handler.Failed,
+				Message:         "error creating cloud provider snapshot restore job: `target_project_id` must be set if delivery type is `automated`",
+				ResourceModel:   currentModel,
+			}, nil
+		}
+	}
+	requestParameters := &matlasClient.SnapshotReqPathParameters{
```
Comment (thetonymaster) [inline cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go]: you can move this to line 27 and then reuse it.

File: cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+		GroupID:     *currentModel.ProjectId.Value(),
+		ClusterName: *currentModel.ClusterName.Value(),
+	}
+	snapshotRequest := &matlasClient.CloudProviderSnapshotRestoreJob{
+		SnapshotID:   *currentModel.SnapshotId.Value(),
+		DeliveryType: *deliveryType,
+	}
+	if *deliveryType == "automated" {
```
Comment (thetonymaster) [inline cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go]: if you move tove Line 45 and 46 before the same comparation against automated you can do this inside the same if

Comment (thetonymaster) [reply cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go]: Also, you might want to use a constant.

File: cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	projectId := *currentModel.ProjectId.Value()
+	snapshotRequest := &matlasClient.SnapshotReqPathParameters{
+		GroupID:     projectId,
+		ClusterName: *currentModel.ClusterName.Value(),
+	}
+	restoreJobs, _, err := client.CloudProviderSnapshotRestoreJobs.List(context.Background(), snapshotRequest)
```
Comment (thetonymaster) [inline cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go]: add a nl

File: cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+		}
+	}
+	requestParameters := &matlasClient.SnapshotReqPathParameters{
+		GroupID:     *currentModel.ProjectId.Value(),
+		ClusterName: *currentModel.ClusterName.Value(),
+	}
+	snapshotRequest := &matlasClient.CloudProviderSnapshotRestoreJob{
+		SnapshotID:   *currentModel.SnapshotId.Value(),
+		DeliveryType: *deliveryType,
+	}
```
Comment (thetonymaster) [inline cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go]: add a nl

File: cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	deliveryType := currentModel.DeliveryType.Value()
+	if deliveryType == nil || (*deliveryType != "automated" && *deliveryType != "download") {
```
Comment (shum) [inline cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go]: Agree with @thetonymaster that `"automated"` and `"download"` should be both made constants

File: cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+		return handler.ProgressEvent{
+			OperationStatus: handler.Failed,
+			Message:         "error creating cloud provider snapshot restore job: you need to implement only one, `automated` and `download` delivery types",
```
Comment (shum) [inline cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go]:
```suggestion
			Message:         "error creating cloud provider snapshot restore job: you must specify either `automated` or `download` delivery types",
```

File: cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Read complete",
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+// Update handles the Update event from the Cloudformation service.
+func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	// operation not available :(
```
Comment (shum) [inline cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go]:
```suggestion
	// NO-OP
```

File: cloud-provider-snapshot-restore-jobs/mongodb-atlas-cloudprovidersnapshotrestorejobs.json [hunk 8]
```diff
@@ -0,0 +1,136 @@
+{
+    "typeName": "MongoDB::Atlas::CloudProviderSnapshotRestoreJobs",
+    "description": "This resource allows to create, cancel, get one or list of cloud provider snapshot restore jobs.",
```
Comment (shum) [inline cloud-provider-snapshot-restore-jobs/mongodb-atlas-cloudprovidersnapshotrestorejobs.json]:
```suggestion
    "description": "This resource allows you to create, cancel, get one or list all cloud provider snapshot restore jobs.",
```

**Pull request merged**